net/ddns-scripts/files/usr/lib/ddns/update_one_com.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/sh
+
+# ONE.COM DDNS SCRIPT
+# REQUIRES CURL
+# $ opkg install curl
+
+# SCRIPT BY LUGICO
+# CONTACT: main@lugico.de
+
+[ -z "$CURL" ] && [ -z "$CURL_SSL" ] && write_log 14 "one.com communication require cURL with SSL support. Please install"
+[ -z "$domain" ]   && write_log 14 "Service section not configured correctly! Missing 'domain'"
+[ -z "$username" ] && write_log 14 "Service section not configured correctly! Missing 'username'"
+[ -z "$password" ] && write_log 14 "Service section not configured correctly! Missing 'password'"
+
+. /usr/share/libubox/jshn.sh
+
+write_log 0 "one.com ddns script started"
+
+local __SUBDOMAIN __MAINDOMAIN __LOGINURL __RECORDID
+local __TTL=3600
+
+COOKIEJAR=$(mktemp /tmp/one_com_cookiejar.XXXXXX) || exit 1
+
+__SUBDOMAIN=$(echo $domain | sed -e 's/[^\.]*\.[^\.]*$//' -e 's/\.$//' )
+__MAINDOMAIN=$(echo $domain | sed -e "s/${__SUBDOMAIN}\.//" )
+
+
+# LOGGING IN
+# GET LOGIN POST URL FROM FORM
+__LOGINURL=$( $CURL \
+	-RsSL \
+	--stderr $ERRFILE \
+	-c $COOKIEJAR \
+	"https://www.one.com/admin/" \
+	| grep 'Login-form login autofill' \
+	| sed -e 's/.*action="//' -e 's/".*//' -e 's/\&amp;/\&/g' \
+)
+
+# POST LOGIN DATA
+$CURL \
+	-RsSL \
+	--stderr $ERRFILE \
+	-c $COOKIEJAR \
+	-b $COOKIEJAR \
+	"${__LOGINURL}" \
+	-H "Content-Type: application/x-www-form-urlencoded" \
+	-X POST \
+	-d "username=${username}&password=${password}&credentialId=" \
+	| grep "Invalid username or password." > $DATFILE
+
+if [ "$?" == "0" ] ; then
+	write_log 14 "Invalid credentials"
+	return 1
+fi
+
+
+# SETTING DOMAIN
+$CURL -RsSL \
+	--stderr $ERRFILE \
+	-c $COOKIEJAR \
+	-b $COOKIEJAR \
+	"https://www.one.com/admin/select-admin-domain.do?domain=${__MAINDOMAIN}" \
+	| grep "<meta name=\"one.com:active-domain\" content=\"${__MAINDOMAIN}\"/>" > $DATFILE
+
+if [ "$?" != "0" ] ; then
+	write_log 14 "Failed to select domain '${__MAINDOMAIN}'"
+	return 1
+fi
+
+
+# GETTING RECORD ID
+records=$( $CURL \
+	-RsSL \
+	--stderr $ERRFILE \
+	-c $COOKIEJAR \
+	-b $COOKIEJAR \
+	"https://www.one.com/admin/api/domains/${__MAINDOMAIN}/dns/custom_records"
+)
+
+json_load "$records"
+
+if json_is_a "result" "object" && \
+	json_select "result" && \
+	json_is_a "data" "array"
+then
+	json_select "data"
+	i=1
+	while json_is_a ${i} "object" ; do
+		json_select "${i}"
+		json_select "attributes"
+		json_get_var "prefix" "prefix"
+		json_close_object
+		if [ "$prefix" == "$__SUBDOMAIN" ] ; then
+			json_get_var "__RECORDID" "id"
+			write_log 0 "Found record id : ${__RECORDID}"
+			break
+		fi
+		json_close_object
+		i=$(($i + 1))
+	done
+fi
+
+
+
+if [ "${__RECORDID}" == "" ] ; then
+	write_log 14 "domain record not found"
+	return 1
+fi
+
+
+
+# SENDING PATCH
+$CURL \
+	-RsSL \
+	--stderr $ERRFILE \
+	-c $COOKIEJAR \
+	-b $COOKIEJAR \
+	-X PATCH \
+	-d "{\"type\":\"dns_service_records\",\"id\":\"${__RECORDID}\",\"attributes\":{\"type\":\"A\",\"prefix\":\"${__SUBDOMAIN}\",\"content\":\"${__IP}\",\"ttl\":${__TTL}}}" \
```

Comment:
I have also noticed that we could use `json_*` there.
https://github.com/openwrt/openwrt/blob/ea49690ff47d14ee096d7c1368cff03f1cce1d76/package/network/config/netifd/files/lib/netifd/dhcp.script#L81

Everything else is ready to merge

Reply:
Done. Patch request is now being generated using jshn.